conans/client/tools/settings.py

```diff
@@ -1,77 +1,382 @@
-            raise ConanInvalidConfiguration("Could not detect the current default cppstd for "
-                                            "the compiler {}-{}.".format(compiler,
-                                                                         compiler_version))
-        return cppstd
+    current_cppstd = deduced_cppstd(conanfile)
+    if not current_cppstd:
+        raise ConanInvalidConfiguration("Could not detect default cppstd for the current compiler.")
 
-    current_cppstd = deduced_cppstd()
-    check_required_gnu_extension(current_cppstd)
+    if gnu_extensions:
+        check_gnu_extension(current_cppstd)
 
-    if less_than(current_cppstd, cppstd):
-        raise ConanInvalidConfiguration("Current cppstd ({}) is lower than the required C++ "
+    if normalized_cppstd(current_cppstd) > normalized_cppstd(cppstd):
+        raise ConanInvalidConfiguration("Current cppstd ({}) is higher than the required C++ "
                                         "standard ({}).".format(current_cppstd, cppstd))
 
 
-def valid_min_cppstd(conanfile, cppstd, gnu_extensions=False):
-    """ Validate if current cppstd fits the minimal version required.
+def valid_max_cppstd(conanfile, cppstd, gnu_extensions=False):
+    """ Validate if current cppstd fits the maximal version required.
 
-    :param conanfile: ConanFile instance with cppstd to be compared
+    :param conanfile: ConanFile instance with cppstd information
     :param cppstd: Minimal cppstd version required
-    :param gnu_extensions: GNU extension is required (e.g gnu17). This option ONLY works on Linux.
+    :param gnu_extensions: GNU extension is required (e.g gnu17)
     :return: True, if current cppstd matches the required cppstd version. Otherwise, False.
     """
+    if not str(cppstd).isdigit():
+        raise ConanException("cppstd parameter must be a number")
+    if not isinstance(gnu_extensions, bool):
+        raise ConanException("gnu_extensions parameter must be a bool")
+
     try:
-        check_min_cppstd(conanfile, cppstd, gnu_extensions)
+        check_max_cppstd(conanfile, cppstd, gnu_extensions)
     except ConanInvalidConfiguration:
         return False
     return True
+
+
+def check_cppstd(conanfile, min=None, max=None, excludes=[], gnu_extensions=False, strict=False):
+    """ Check if the current cppstd fits specified requirements
+
+        In case the current cppstd doesn't fit specified requirements
+        a ConanInvalidConfiguration exception will be raised.
+
+        In case information about cppstd is lacking
+        a ConanUnknownConfiguration exception will be raised.
+
+    :param conanfile: ConanFile instance
+    :param min: Minimal cppstd version required
+    :param max: Maximal cppstd version required
+    :param excludes: A list of cppstd version excluded
+    :param gnu_extensions: GNU extension is required (e.g gnu17)
+    :param strict: Unkown configurations are invalid
+    """
+    if min and not str(min).isdigit():
```

Comment:
I don't think we need so many so strict type checks everywhere (given the duck typing, it's not a very pythonic way for my point of view).

Reply:
The point is that these are user-facing functions and users are not necessary familiar with the Python philosophy. I'm not necessary against removing them, but I'd prefer to have clearer error messages for users other than a python stack trace.

Reply:
usually I don't like many checks, but as those methods a public for users, they can pass anything, so I agree adding these messages.